src/crangler/c_wrangler.cpp

```diff
@@ -449,7 +452,10 @@ static void mangle_function(
         {
           for_count++;
           const auto &invariant =
-            loop_invariants["for" + std::to_string(for_count)];
+            loop_invariants.count("for" + std::to_string(for_count))
+              ? loop_invariants["for" + std::to_string(for_count)]
+              : loop_invariants
+                  ["loop" + std::to_string(while_count + for_count)];
```

Comment:
Just checking: the code below this line generates "invariant(" ... while the code above generates "loop_invariant(" ... - are both of these correct?

Reply:
No, only ```loop_invariant``` is correct. I will make the tests cover this bugs and create another PR to fix the bugs

Reply:
The bug is fixed in the [PR#7107](https://github.com/diffblue/cbmc/pull/7107).